### ISSUE TITLE
Routes aren't updated when new Resource is broadcast to connlib

### ISSUE THREAD

**Issue by jamilbk** [CloudFire-LLC/cloudfire#4209]
just tested this:

- Add a policy for a resource
- Client doesn't receive updated route for resource

**Comment by jamilbk** [comment]
The route is removed properly, just not added. Here is a sequence of printing route after each step of `Sign in` -> `Disable policy` -> `Enable policy`

```
jamil@Airbook-Mac:~ % netstat -rn | grep 10.0.0.83
10.0.0.83/32       link#49            UCS            utun12       
jamil@Airbook-Mac:~ % netstat -rn | grep 10.0.0.83
jamil@Airbook-Mac:~ % netstat -rn | grep 10.0.0.83
jamil@Airbook-Mac:~ % 
```

The last `netstat` should have printed the Resource in my routing table.

**Comment by conectado** [comment]
Fixed by  #4211
